ddtrace/_events.py

```diff
@@ -0,0 +1,98 @@
+from typing import List
+from typing import Mapping
+from typing import Optional
+from typing import TYPE_CHECKING
+from typing import Union
+
+from ._hooks import Hooks
+
+
+if TYPE_CHECKING:
+    from ddtrace.span import Span
+
+
+_HTTP_HOOKS = Hooks()
+
+REQUEST = "request"
+RESPONSE = "response"
+
+
+def emit_http_request(
```

Comment:
What about having something more generic such as:
`emit_event(event)`

where event would be a class such as `HTTPEvent` like:
```
class Event:
    span: Span
    integration: str

class HTTPEvent(Event):
    method: str
    url: str
    …
```

That'd make the system reusable for other events.

You can then use a `register_event(event_cls)` which register function for particular event types.

Reply:
I like it! It also comes with a nice to have feature: we can now add validators to attributes for unit testing 🥳